rust/perspective-python/Cargo.toml

```diff
@@ -52,9 +52,12 @@ pyo3-build-config = "0.20.2"
 python-config-rs = "0.1.2"
 
 [dependencies]
+# NOTE: when building from the git repo, these perspective-* dependencies are
+# overridden with path dependencies in .cargo/config.toml.  This is done to
+# support the sdist, which doesn't include these packages.
```

Comment:
there may be pitfalls in doing it this way I'm not aware of, but it seemed less messy than patching the Cargo.toml file in the tarball.

Reply:
Since the new test_python_sdist job in CI runs in a git checkout, the path directives in .cargo/config.toml will apply when that job builds the perspective-python crate (and wheel) from the sdist. That means in test_python_sdist, changes in the git repo to perspective-client and perspective-server which haven't yet been published to crates.io would be tested too. 

End users building from that same sdist artifact, if it's from between releases, would get a different result, building against the most recent release of those crates on crates.io. Workarounds for them would be to set up a `.cargo/config.toml` (maybe one in ~? I don't know if pip's build isolation would defeat that) which uses paths pointing into a local perspective git clone. This is an unfortunate downside of bundling only perspective-python's source in the sdist

Reply:
Oh, uh, I stand corrected. The sdist build in CI is downloading from crates.io https://github.com/tomjakubowski/perspective-oneoff-tests/actions/runs/11604564693/job/32313519748#step:8:187

I think that's because of pip's build isolation -- the build is happening in some other working directory, so Cargo doesn't find .cargo/config.toml

will want to adjust the CI job. I think, like I mentioned setting it in `$HOME/.cargo/config.toml` will work